### PR TITLE
Fix 915 - Revert to Microsoft.Azure.Cosmos v3

### LINKED ISSUE #915: API: Use `Microsoft.Azure.Cosmos` library instead of `Azure.Cosmos-preview`
`Azure.Cosmos Version=4.0.0-preview3` has not been updated since 2020 and is pointing to the same github repository as `Microsoft.Azure.Cosmos Version=3.23.0`

Additionally, `Azure.Cosmos Version=4.0.0-preview3` is vulnerable to [CVE-2021-26701](https://www.cve.org/CVERecord?id=CVE-2021-26701)

**Suggested change**
Use `Microsoft.Azure.Cosmos Version=3.23.0` instead of `Azure.Cosmos Version=4.0.0-preview3`

### PR DESCRIPTION
**Pull request by steinsiv**
## Fixes
This pull request fixes #915 

## Description
https://www.nuget.org/packages/Azure.Cosmos
No updates in two years, downgrading to v3

## Type of change
Change of library

* Enhancement of existing functionality

## Impacted Areas in Application
Azure Cosmos connections

## Checklist:

*Communication*
* [x] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs [hunk 1]
```diff
@@ -35,7 +34,7 @@ public async Task<TDocument> GetDocumentAsync(TDocumentId id)
             var container = cosmosClient.GetContainer(dbName, containerId);
             var itemResponse = await container.ReadItemAsync<TDocument>(id.ToString(), new PartitionKey(id.ToString()));
```
Comment (lindsve) [inline Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs]: Might as well return the `itemResponse` directly without storing it as a variable

File: Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs [hunk 2]
```diff
@@ -55,26 +56,26 @@ await foreach (var document in container.GetItemQueryIterator<TDocument>(queryDe
         public async Task<TDocument> UpdateDocumentAsync(TDocumentId id, TDocument document)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
-            var itemResponse = await container.ReplaceItemAsync(document, document.Id.ToString());
-            return itemResponse.Value;
+            var itemResponse = await container.ReplaceItemAsync<TDocument>(document, document.Id.ToString());
```
Comment (lindsve) [inline Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs]: Can return directly here as well

File: Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs [hunk 3]
```diff
@@ -55,26 +56,26 @@ await foreach (var document in container.GetItemQueryIterator<TDocument>(queryDe
...
+            return itemResponse;
         }
 
         public async Task<TDocument> CreateDocumentAsync(TDocument document)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
-            var createResponse = await container.CreateItemAsync(document);
+            var createResponse = await container.CreateItemAsync<TDocument>(document, new PartitionKey(document.Id.ToString()));
```
Comment (lindsve) [inline Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs]: Return directly

File: Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs [hunk 4]
```diff
@@ -55,26 +56,26 @@ await foreach (var document in container.GetItemQueryIterator<TDocument>(queryDe
...
+            return createResponse;
         }
 
         public async Task DeleteDocumentAsync(TDocumentId id)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
 
             var deleteResponse = await container.DeleteItemStreamAsync(id.ToString(), new PartitionKey(id.ToString()));
-            if (deleteResponse.Status != 204)
+            if (((uint) deleteResponse.StatusCode) != 204)
```
Comment (lindsve) [inline Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs]: `if (deleteResponse.StatusCode != HttpStatusCode.NoContent)`

Comment (steinsiv) [reply Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs]: Much better, thanks 👍

File: Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs [hunk 5]
```diff
@@ -55,26 +55,23 @@ await foreach (var document in container.GetItemQueryIterator<TDocument>(queryDe
...
         public async Task DeleteDocumentAsync(TDocumentId id)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
 
             var deleteResponse = await container.DeleteItemStreamAsync(id.ToString(), new PartitionKey(id.ToString()));
-            if (deleteResponse.Status != 204)
+            if (deleteResponse.StatusCode != HttpStatusCode.NoContent)
             {
-                throw new RepositoryException($"Unable to delete document with id: {id}", deleteResponse.Status);
+                throw new RepositoryException($"Unable to delete document with id: {id}", (int) deleteResponse.StatusCode);
```
Comment (lindsve) [inline Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs:74]: The comment suggests that it is the document id that is printed (`Unable to delete document with id: `), but it is the statusCode that is placed here. Seems to be an old issue, but I suggest replacing `deleteResponse.StatusCode` with `id` here

**Review by lindsve — APPROVED** [review]
- [x] Read issue and PR
- [ ] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**